src/Symfony/Component/Cache/Marshaller/PhpMarshaller/Marshaller.php

```diff
@@ -0,0 +1,210 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Cache\Marshaller\PhpMarshaller;
+
+/**
+ * @author Nicolas Grekas <p@tchwork.com>
+ *
+ * @internal
+ */
+class Marshaller
+{
+    /**
+     * Prepares an array of values for PhpMarshaller.
+     *
+     * For performance this method is public and has no type-hints.
+     *
+     * @param array             &$values
+     * @param \SplObjectStorage $objectsPool
+     * @param array             &$refsPool
+     * @param int               &$objectsCount
+     * @param bool              &$valuesAreStatic
+     *
+     * @return int
+     *
+     * @internal
+     */
+    public static function marshall($values, $objectsPool, &$refsPool, &$objectsCount, &$valuesAreStatic)
```

Comment:
should we have typehints here ?

Reply:
See above:
> For performance this method is public and has no type-hints. 

Since this method (and class) is internal, we can do it.